### ISSUE TITLE
Rounding setting not correctly applied with midpoints

### ISSUE THREAD

**Issue by mazacm01** [Seddryck/NBi#311]
Hi, I'm  trying to compare two percentage values, one from MS SSAS cube and one from MS SQL database. I applied this rounding setting: 

![image](https://user-images.githubusercontent.com/33053372/31931666-3695fe80-b8a4-11e7-869d-ceaaf0c164be.png)

But the test result ends like this:

![image](https://user-images.githubusercontent.com/33053372/31931543-d1a42c54-b8a3-11e7-9ba0-3f9dd1ef4483.png)

Do you have any thoughts why round is not working?
THX

**Comment by Seddryck** [comment]
Rounding is rarely the good choice, usually tolerance is a better choice. In next versions this feature (rounding) should be moved to the transformations of a result-set in place of being associated to the tolerance.

A rounding is applied to the actual result-set not to the expected result-set. Let me guess that in your case the result-set defined in the assert is returning 19.15 and the query defined in the system-under-test, returns 19.19 (rounded to 19.2) ... 19.15 is still different than 19.2.

Or you should apply a ```ROUND``` in your SQL query or switch to tolerance.

**Comment by mazacm01** [comment]
OK, sorry. I assumed from your documentation that rounding is applied to both values.

![image](https://user-images.githubusercontent.com/33053372/31932484-b364f4b4-b8a6-11e7-8157-97263290273b.png)

**Comment by Seddryck** [comment]
Huuuuum ... will need to double-check this, it has been implemented years ago and I use it in rare cases. The documentation is a justification of the implemention as a comparison tool and not a transformation tool ... could be a bug. But will need to check this further.

**Comment by Seddryck** [comment]
You're 100% correct. The rounding should be applied to both result-sets. You should disregard my first comment.

Will investigate a bug, it could be two things: 
* one of the result-set is not rounded (doubtful) 
* an issue that -19.15 is rounded to -19.1 and not -19.2 (higher probability - is it really an issue or is it normal rounding rules?)


Remarks for @seddryck: 
* DateTimeComparer and NumericComparer have no tests with Rounding.
* The behavior of rounding the middle point is not asserted in [NBi.Testing.Unit.Core.ResultSet.Comparer.NumericRoundingTest](https://github.com/Seddryck/NBi/blob/96fb634264549cbfa34840723ebac39a35da2b23/NBi.Testing/Unit/Core/ResultSet/Comparer/NumericRoundingTest.cs)

**Comment by Seddryck** [comment]
I just checked right now and ```Math.Round(-19.15, 1)``` is equal to ```-19.2``` and my code is returning ```-19.1```. Definitively not inline with documentation where I explicitly said that we're replicating the behavior of .Net framework

### Will fix it this week.
- [x]  in [NBi.Core.ResultSet.Comparer.NumericRounding](https://github.com/Seddryck/NBi/blob/96fb634264549cbfa34840723ebac39a35da2b23/NBi.Core/ResultSet/Comparer/NumericRounding.cs) stop to convert into double (Was I drunk?) !!! Stay in decimal.
- [x] in [NBi.Core.ResultSet.Comparer.Rounding](https://github.com/Seddryck/NBi/blob/96fb634264549cbfa34840723ebac39a35da2b23/NBi.Core/ResultSet/Comparer/Rounding.cs) change ```>``` by ```>=``` for Round in GetValue() ... refactor this method to move it to NumericRounding
- [x]  add all missing tests (see above)

**Comment by Seddryck** [comment]
Math.Round is supporting an additional parameter with to determine what to do with the mid-point. The second implementation is giving the following results:
```c-sharp
* Math.Round(19.15, 1, MidpointRounding.ToEven)    // 19.2
* Math.Round(19.25, 1, MidpointRounding.ToEven)    // 19.2
```
Is it what you're looking for (check twice the results and I don't made any typo)? If yes I could implement an additional parameter for the rounding to specify this behavior. If not could you describe a few examples so I can implement the way you want to round?

**Comment by Seddryck** [comment]
There is an additional debate about should it report ```19.15<>19.2``` or ```19.1<>19.2```. Feel free to express your preference and give some arguments in favor of one view or the other.

**Comment by Seddryck** [comment]
A fix is available (targeting v1.16) at https://ci.appveyor.com/project/Seddryck/nbi/build/1.16.0-beta.89.build.100

If you want to reopen the debate on the rendering when the test is failing, feel free to create a new issue.
